EventListener/CsrfListener.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\EventListener;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Csrf;
+use Sensio\Bundle\FrameworkExtraBundle\Exception\InvalidCsrfTokenException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+/**
+ * CsrfListener handles csrf restrictions on actions.
+ *
+ * @author Konstantin Myakshin <koc-dp@yandex.ru>
+ */
+class CsrfListener implements EventSubscriberInterface
+{
+    private $csrfTokenManager;
+
+    public function __construct(CsrfTokenManagerInterface $csrfTokenManager = null)
```

Comment:
Allowing `null` here makes it possible to create the listener in an unusable state. I think you should rather remove (or not create) the definition instead when no CSRF token manager is available (see also my comment below).